### PR TITLE
add paperwallet glue code

### PR DESCRIPTION
**Pull request by vincenthz**

### CODE REVIEW

File: js/PaperWallet.js [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+export const scramble = (module, iv, password, input) => {
+    if (iv.length != 4) {
+        throw new Error('IV must be 4 bytes');
+    }
+    const bufiv = newArray(module, iv);
+    const bufinput = newArray(module, input);
+    const bufpassword = newArray(module, password);
+    const bufoutput = newArray0(module, input.length + 4);
+    module.paper_scramble(bufiv, bufpassword, password.length, bufinput, input.length, bufoutput);
+    let result = copy_array(module, bufoutput, input.length + 4);
```
Comment (DominikGuzei) [inline js/PaperWallet.js]: `copy_array` will not work here, since it's now called `copyArray`

File: js/PaperWallet.js [hunk 2]
```diff
@@ -1,4 +1,40 @@
+import { newArray, newArray0, copyArray } from './utils/arrays';
```
Comment (DominikGuzei) [inline js/PaperWallet.js]: We don't have `newArray0` in the utils … i refactored it to a single `newArray` method that can be called with `true` as last param to indicate that it should nullify.

Comment (vincenthz) [reply js/PaperWallet.js]: newArray0 take an implicit size, not assume it from the input parameter. it doesn't work in this case. (one might say that newArray was miscalled in the first place and should be a copyArray like name)

File: js/PaperWallet.js (hunk 1 above)
Comment (vincenthz) [reply js/PaperWallet.js]: good catch. looking forward to be able to test everything smoothly

File: js/utils/arrays.js [hunk 3]
```diff
@@ -9,6 +9,16 @@ export const newArray = (module, b, isZero=false) => {
   return ptr
 }
 
+export const newArray0 = (module, len) => {
```
Comment (DominikGuzei) [inline js/utils/arrays.js]: Haha, ok … you really want this duplicated code i guess ;) please look at the `newArray` function, then you will see what i meant … this is already implemented there

File: js/PaperWallet.js (hunk 2 above)
Comment (DominikGuzei) [reply js/PaperWallet.js]: Ah ok… sorry then i misunderstood how it works … although the implementation looked 99% the same apart from this single param

File: js/utils/arrays.js (hunk 3 above)
Comment (vincenthz) [reply js/utils/arrays.js]: how do you expect `b.length` work when b doesn't exist ?

**Pull request merged**